tests/Microsoft.Identity.Test.Integration.netfx/SeleniumTests/InteractiveFlowTests.cs

```diff
@@ -212,31 +212,51 @@ public async Task ValidateCcsHeadersForInteractiveAuthCodeFlowAsync()
 
             Assert.IsNotNull(authResult);
             Assert.IsNotNull(authResult.AccessToken);
+        }
+
+        [TestMethod]
+        public async Task Interactive_Arlington_MultiCloudSupport_AADAsync()
+        {
+            // Arrange
+            LabResponse labResponse = await LabUserHelper.GetArlingtonUserAsync().ConfigureAwait(false);
+            var tuple = getPCA(labResponse, false, true);
+            IPublicClientApplication pca = tuple.Item1;
+
+            Trace.WriteLine("Part 1 - Acquire a token interactively");
+            AuthenticationResult result = await pca
+                .AcquireTokenInteractive(s_scopes)
+                .WithCustomWebUi(CreateSeleniumCustomWebUI(labResponse.User, Prompt.SelectAccount, false))
+                .ExecuteAsync(new CancellationTokenSource(_interactiveAuthTimeout).Token)
+                .ConfigureAwait(false);
+
+            Assert.IsNotNull(result);
+            Assert.IsNotNull(result.Account);
+            Assert.AreEqual(labResponse.Lab.Authority + labResponse.Lab.TenantId, result.Account.GetAuthority());
+
+            Trace.WriteLine("Part 2 - Get Accounts");
+            var accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(accounts);
+            Assert.IsTrue(accounts.Count() > 0);
+            Assert.AreEqual(labResponse.User.Upn, accounts.FirstOrDefault().Username);
```

Comment:
I think there might be a bug?
GetAccounts returns the account with the global environment. While ATS and ATI return account correctly with national environment.

Reply:
@neha-bhargava  - please add some more assertions around the account objects. IN particular, I think the accountId and ENV should be equal between the accoutn from ATI and GetAccounts()

@pmaytak - `GetAccounts` returns all the accounts, irrespective of cloud, when the "MultiCloudSupport" flag is enabled. So I think in this case `GetAccounts` returns the national cloud account. Can you give more details about where you see the bug?

Reply:
Repro (can use NetCoreWinFormsWMA, just uncheck `Use WAM`):
- Create PCA, add `WithMultiCloudSupport`, use global authority.
- Call AcquireTokenInteractive, login with US account.
   - AuthenticationResult.Account.Environment is national.
 - Call AcquireTokenSilent, Add `WithAuthority(result.acc.GetAuthority())`.
   - AuthResult.Account.Environment is still national.
 - Call GetAccounts
   - Returns one account, but Account.Environment is global.
 - Call GetAccount with result.Account.HomeAccountId.Identifier.
   - Also returns account with global environment.

All tokens and accounts are saved with national environment in the cache.
I think the issue might be here:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/a65f7afa4eabbfc04d6696a1f6c1ce430c3da8a9/src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs#L989

Reply:
Ok got it. Thanks for the explanation @pmaytak.

Reply:
Marking this as resolved since we are not using GetAuthority any more.

Reply:
GetAccount/GetAccounts still returns account with global environment, not national.
Repro for example. Passing `interactiveResult.Account` into ATS will work, passing `account`, does not work.
```
            var interactiveResult = await pca.AcquireTokenInteractive(new[] { scopeGraph })
                                .WithUseEmbeddedWebView(true)
                                .ExecuteAsync();

            var account = await pca.GetAccountAsync(result.Account.HomeAccountId.Identifier);

            var result1 = await pca.AcquireTokenSilent(new[] { scopeGraph }, account)
                .ExecuteAsync();
```